File: backend/app/api/v1/endpoints/health.py
from fastapi import APIRouter

from app.db import check_db_connection
from app.schemas.envelope import ApiEnvelope

router = APIRouter(tags=["health"])


@router.get("/health", response_model=ApiEnvelope[dict[str, str | bool]])
def health() -> ApiEnvelope[dict[str, str | bool]]:
    """返回服务健康状态，并用最小查询确认数据库可达。"""
    db_ok = check_db_connection()
    return ApiEnvelope(
        data={
            "status": "ok" if db_ok else "degraded",
            "database": db_ok,
        }
    )

File: apps/api/src/routes/v1/health.ts
import { OpenAPIHono, createRoute, z } from "@hono/zod-openapi";
import { checkDbConnection } from "@feedmind/db";
import { successEnvelope } from "../../lib/openapi-schemas.js";

export const healthRoutes = new OpenAPIHono();

const healthDataSchema = z.object({
  status: z.enum(["ok", "degraded"]).describe("服务状态"),
  database: z.boolean().describe("数据库连接是否正常"),
});

const healthRoute = createRoute({
  method: "get",
  path: "/health",
  responses: {
    200: {
      content: { "application/json": { schema: successEnvelope(healthDataSchema) } },
      description: "健康检查 — 返回服务及数据库状态",
    },
    500: {
      content: {
        "application/json": {
          schema: z.object({
            data: z.null(),
            error: z.object({ code: z.string(), message: z.string() }),
          }),
        },
      },
      description: "服务器内部错误",
    },
  },
});

healthRoutes.openapi(healthRoute, async (c) => {
  try {
    const database = await checkDbConnection();
    return c.json(
      {
        data: { status: database ? ("ok" as const) : ("degraded" as const), database },
        error: null,
      },
      200,
    );
  } catch {
    return c.json({ data: null, error: { code: "INTERNAL_ERROR", message: "健康检查失败" } }, 500);
  }
});
